bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -468,12 +478,16 @@ public int compare(EntryLogMetadata m1, EntryLogMetadata m2) {
     public void shutdown() throws InterruptedException {
         this.running = false;
         LOG.info("Shutting down GarbageCollectorThread");
-        if (compacting.compareAndSet(false, true)) {
-            // if setting compacting flag succeed, means gcThread is not compacting now
-            // it is safe to interrupt itself now
-            this.interrupt();
+
+        while (!compacting.compareAndSet(false, true)) {
+            // Wait till the thread stops compacting
+            wait(1000);
```

Comment:
@govind-menon maybe we can replace the "wait(1000)" with a Thread.sleep(100), the '100' instead of '1000' will lead to faster shutdown. does it sound good to you ?